Configure ECSPI for phyBOARD Polis

pefech opened the pull request:
This PR adds ECSPI Support for the imx8m SoCs and enables it for the phyBOARD Polis.
The implementation was tested using the SPI loopback test. 

The documentation for the phyBOARD Polis was updated accordingly.

pefech committed:
dts: arm: nxp: nxp_imx8m_m4: Add ECSPI devices

Add device tree instances for ECSPI devices,
update SoC code to enable clocks.

Signed-off-by: Peter Fecher <p.fecher@phytec.de>

pefech committed:
boards: arm: phyboard_polis: configure spi and can

Configures ECSPI devices in devicetree including connected
mcp2518 (CAN FD).
Configures pinctrl for board.
Enables spi support in board files.

Signed-off-by: Peter Fecher <p.fecher@phytec.de>

pefech committed:
tests: drivers: spi_loopback: Add phyboard_polis

Adds the devicetree overlay and configuration for
phyboard polis to the spi_loopback test.

Signed-off-by: Peter Fecher <p.fecher@phytec.de>

pefech committed:
boards: phytec: update phyboard_polis docs

Updating PhyBOARD Polis docs after enabling ECSPI
and CAN support on that board.
Add additional information what to do if
using Linux and Zephyr simultaneously.

Signed-off-by: Peter Fecher <p.fecher@phytec.de>

pefech commented:
This is the output of the SPI Loopback Test:
```
*** Booting Zephyr OS build v3.7.0-1752-g7b4e7ff9eda0 ***
Running TESTSUITE spi_loopback
===================================================================
START - test_spi_loopback
I: SPI test on buffers TX/RX 0x200041e0/0x200041c0, frame size = 8
I: SPI test slow config
I: Start complete multiple
I: Passed
I: Start complete loop
I: Passed
I: Start null tx
I: Passed
I: Start half start
I: Passed
I: Start half end
I: Passed
I: Start every 4
I: Passed
I: Start rx bigger than tx
I: Passed
I: Start complete large transfers
I: Passed
I: Start async call
I: Passed
I: SPI test fast config
I: Start complete multiple
I: Passed
I: Start complete loop
I: Passed
I: Start null tx
I: Passed
I: Start half start
I: Passed
I: Start half end
I: Passed
I: Start every 4
I: Passed
I: Start rx bigger than tx
I: Passed
I: Start complete large transfers
I: Passed
I: Start async call
I: Passed
I: Start complete loop
I: Passed
I: Start complete loop
I: Passed
I: All tx/rx passed
 PASS - test_spi_loopback in 0.513 seconds
===================================================================
TESTSUITE spi_loopback succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [spi_loopback]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.513 seconds
 - PASS - [spi_loopback.test_spi_loopback] duration = 0.513 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL
```

JiafeiPan commented on dts/arm/nxp/nxp_imx8m_m4.dtsi:
```diff
@@ -178,6 +178,36 @@
 			status = "disabled";
 		};
+
+		ecspi1: spi@30820000 {
```
Would you please reorder the nodes in address?

pefech replied:
Done.
Maybe this should also be done for `nxp_imx8ml_m7.dtsi`? There are nodes at the end that are not ordered in addresses.

JiafeiPan replied:
yes, we should, but this PR has no relation with it, unless a PR to update it.

JiafeiPan commented on dts/arm/nxp/nxp_imx8m_m4.dtsi:
```diff
@@ -180,7 +210,7 @@
 	};
 };
-
+https://github.com/zephyrproject-rtos/zephyr/pull/77736
```
what is it for?

pefech replied:
Whoops, this shouldn't be there. Interesting, that all checks passed

JiafeiPan commented on tests/drivers/spi/spi_loopback/boards/mimx8mm_phyboard_polis_mimx8mm6_m4.overlay:
```diff
@@ -0,0 +1,42 @@
+			bias-pull-up;
+		};
+	};
+};
```
per my understand, pinctrl should be board specified,  but have no relation with test case, right? so move pinctrl property of scspi1 to board dts? so that can reused by other application or test case.

pefech replied:
Yes, this seems better.

pefech commented:
- Moved the pincontrol into the boards devicetree files
- added the ecspi nodes in the board files with the corresponding cs pins and on-board connected devices
- updated documentation furthermore

The added on-board CAN IC was tested with another known working CAN device. It seems like the MCP2518 driver has no possibility to turn on any supply-gpios for the receiver. So the board can only listen at the moment. I want to update the driver in another PR.

I hope adding the CAN device made the PR not messy as this was not initially planned. I updated the commit messages correspondingly.

JiafeiPan commented on tests/drivers/spi/spi_loopback/boards/mimx8mm_phyboard_polis_mimx8mm6_m4.overlay:
```diff
@@ -0,0 +1,30 @@
+&ecspi1 {
+	status = "okay";
+	pinctrl-0 = <&ecspi1_default>;
+	pinctrl-names = "default";
```
remove pinctrl-0 and pinctrl-names as have already in board dts?

pefech replied:
Done

JiafeiPan commented:
Please fix CI issues
